src/input_handlers.h

```diff
@@ -263,15 +225,91 @@ class VmaZCopyReadInputHandler : public MessageParser<BufferAccumulation> {
             return process_buffer(callback, m_recv_data, m_ptr->m_pkt_buf, m_non_zcopy_len);
         }
     }
+};
+
+#ifdef USING_VMA_EXTRA_API // VMA
+class SocketXtremeInputHandler : public MessageParser<BufferAccumulation> {
+private:
+    SocketRecvData &m_recv_data;
+    vma_buff_t *m_vma_buff;
+public:
+    inline SocketXtremeInputHandler(Message *msg, SocketRecvData &recv_data):
+        MessageParser<BufferAccumulation>(msg),
+        m_recv_data(recv_data),
+        m_vma_buff(NULL)
+    {}
+
+    /** Receive pending data from a socket
+     * @param [in] socket descriptor
+     * @param [out] recvfrom_addr address to save peer address into
+     * @param [inout] in - storage size, out - actual address size
+     * @return status code
+     */
+    inline int receive_pending_data(int fd, struct sockaddr *recvfrom_addr, socklen_t &size)
+    {
+        //TODO: update after IPv6 will be supported in libvma
```

Comment:
I think that this comment may be removed since we not plan to add IPv6 into libvma.

Reply:
Done